File: 3-swagger-api-documentation/0-typescript/src/modules/dog/dog.controller.ts
/**
 * REST controller for Dog feature.
 */
import {
    Controller, Get
} from "@nestjs/common"
import {
    ApiOperation, ApiResponse, ApiTags
} from "@nestjs/swagger"
import {
    ResponseMessage
} from "../../common/decorators"
import {
    DogService
} from "./dog.service"
import type {
    DogRecord
} from "./dog.service"

/** `/dogs` surface for a second Swagger tag demonstrating route grouping.
 */
@ApiTags("Dogs Module")
@Controller("dogs")
export class DogController {
    constructor(private readonly dogService: DogService) {}

    /** returns list of demo dogs.
     * array of dog records.
     */
    @Get()
    @ApiOperation({
        summary: "Retrieve all dogs",
    })
    @ApiResponse({
        status: 200,
        description: "List of dogs returned",
    })
Get all dogs success)"
    findAll(): DogRecord[] {
        return this.dogService.findAll()
    }
}

File: 3-swagger-api-documentation/backend/2-csharp/Dog/DogController.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;
using SwaggerDemo.Common;

namespace SwaggerDemo.Dog;

/// <summary>
/// <c>/dogs</c> surface for a second Swagger tag demonstrating route grouping.
/// </summary>
[ApiController]
[Route("dogs")]
[SwaggerTag("Dogs Module")]
public class DogController : ControllerBase
{
    private readonly DogService _dogService;

    /// <summary>
    /// Injects <c>DogService</c> via the constructor.
    /// </summary>
    /// <param name="dogService">dog business service.</param>
    public DogController(DogService dogService) => _dogService = dogService;

    /// <summary>
    /// returns the list of demo dogs.
    /// </summary>
    /// <returns>array of dog records inside the envelope.</returns>
    [HttpGet]
    [SwaggerOperation(Summary = "Retrieve all dogs")]
    [ProducesResponseType(typeof(IReadOnlyList<DogRecord>), StatusCodes.Status200OK)]
    [ResponseMessage("Lấy danh sách chó thành công (EN: Get all dogs success)")]
    public ActionResult<IReadOnlyList<DogRecord>> FindAll()
        => Ok(_dogService.FindAll());
}
